typescript/packages/plugins/minting-api/src/impl.ts

```diff
@@ -0,0 +1,116 @@
+import { z } from "zod";
+import type { Plugin, WalletClient } from "@goat-sdk/core";
+import { randomUUID } from "node:crypto";
+
+export function mintAPI(apiKey: string): Plugin<WalletClient> {
+    return {
+        name: "minting_api",
+        supportsSmartWallets: () => true,
+        supportsChain: (chain) => chain.type === "solana" || chain.type === "evm",
+        getTools: async () => {
+            return [
+                {
+                    name: "create_collection",
+                    description: "This {{tool}} creates an NFT collection and returns the ID of the collection.",
+                    parameters: createCollectionParametersSchema,
+                    method: createCollectionMethod(apiKey),
+                },
+                {
+                    name: "mint_nft",
+                    description:
+                        "This {{tool}} mints an NFT to a recipient from a collection and returns the transaction hash. Requires a collection ID of an already deployed collection.",
```

Comment:
Could we make this take a contract address instead? We support that already iirc

Reply:
I don't think we do. I think we only added support for referencing NFTs by their locator